scripts/compress_images_test.py

```
@@ -0,0 +1,232 @@
+        )
+        result_images = (
+            compressor.find_compressible_images()
+        )
+
+        self.assertGreaterEqual(len(result_images), 1)
+
+        for image in result_images:
+            path_extension = image['path'].suffix.lower()
+            expected_path = (
+                pathlib.Path(self.temp_dir) / f'test_image{path_extension}'
+            )
+            self.assertEqual(image['path'].resolve(), expected_path)
+
+    def test_compress_images(self) -> None:
+        """Test image compression process."""
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        result_images = (
+            compressor.find_compressible_images()
+        )
+
+        self.assertGreaterEqual(len(result_images), 1)
+        compressor.compress_images(result_images)
+        for image in result_images:
+            original_path = image['path']
+            original_size = image['original_size']
+
+            rel_path = os.path.relpath(original_path, self.temp_dir)
+            output_path = pathlib.Path(self.output_dir) / rel_path
+
+            self.assertTrue(output_path.exists())
+            compressed_size = output_path.stat().st_size
+            self.assertLess(compressed_size, original_size)
+
+    def test_compress_images_directory_removal_error(self) -> None:
+        """Test error handling when removing output directory fails."""
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        os.makedirs(self.output_dir, exist_ok=True)
+        result_images: List[CompressedImageInfo] = [{
+            'path': pathlib.Path(self.png_path),
+            'original_size': 1000,
+            'new_size': 500
+        }]
+
+        mock_rmtree = mock.Mock(spec=shutil.rmtree)
+        mock_rmtree.error = PermissionError('Failed to remove directory')
```

My sense is that `shutil.rmtree` will encounter and silently ignore errors mostly because the output directory doesn't exist. Is that correct? If so, it'd be nice for this error to reflect that more common scenario